jhub_apps/hub_client/hub_client.py

```diff
@@ -183,15 +184,19 @@ def _create_server(self, username: str, servername: str, user_options: UserOptio
         logger.info("Creating new server", server_name=servername)
         r = requests.post(API_URL + url, headers=self._headers(), json=data)
         r.raise_for_status()
-        if is_jupyterhub_5():
-            logger.info("Sharing", share_with=user_options.share_with)
-            self._share_server_with_multiple_entities(
-                username,
-                servername,
-                share_with=user_options.share_with
-            )
+        if user_options.framework != Framework.jupyterlab.value:
+            if is_jupyterhub_5():
+                logger.info("Sharing", share_with=user_options.share_with)
+                self._share_server_with_multiple_entities(
+                    username,
+                    servername,
+                    share_with=user_options.share_with
+                )
+            else:
+                logger.info("Not sharing server as JupyterHub < 5.x")
```

Comment:
What is the reasoning behind this restriction?

Reply:
Sharing server is a new feature, only became available in JupyterHub 5.x

Regarding the restriction to share the JupyterLab server is to avoid accidentally sharing the users's full filesystem as that's huge security risk.